usaspending_api/search/v2/views/search.py

```diff
@@ -698,21 +698,30 @@ def post(self, request):
         if model == 'SummaryAwardView':
             queryset = queryset \
                 .values("category") \
-                .annotate(category_count=Sum('counts')) \
-                .exclude(category__isnull=True)
+                .annotate(category_count=Sum('counts'))
         else:
+            # for IDV CONTRACTS category is null. change to contract
             queryset = queryset \
                 .values('category') \
-                .annotate(category_count=Count('category')) \
-                .values('category', 'category_count') \
-                .exclude(category__isnull=True)
+                .annotate(category_count=Count(Coalesce('category', Value('contract')))) \
+                .values('category', 'category_count')
 
         results = {"contracts": 0, "grants": 0, "direct_payments": 0, "loans": 0, "other": 0}
 
+        categories = {
+            'contract': 'contracts',
+            'grant': 'grants',
+            'direct payment': 'direct_payments',
+            'loans': 'loans',
+            'other': 'other'
+        }
+
         # DB hit here
         for award in queryset:
-            result_key = award['category'].replace(' ', '_')
-            result_key += 's' if result_key not in ['other', 'loans'] else ''
+            if award['category'] not in categories.keys():
+                result_key = 'other'
+            else:
+                result_key = categories[award['category']]
             results[result_key] = award['category_count']
```

Comment:
Instead of `=` it should be `+=`, Just having an equal sign replaces the old value with the new one. For instance, the end result for FY 2016 is 
```
"results": {
		"contracts": 1689409,
		"grants": 626707,
		"direct_payments": 1467337,
		"loans": 192531,
		"other": 47707
	}
```
Other should equal 47707+ 611535, 47707 is just the None value

Reply:
will fix